Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,342 @@
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    protected $format = AsyncQueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var array
+     */
+    protected $amqpConnections = [];
+
+    /**
+     * Number of the port used by the broker
+     *
+     * @var int $port
+     */
+    protected $port;
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
+     *
+     * {@inheritDoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $this->connectionsData[] = [
+            'host' => $host,
+            'user' => $username,
+            'password' => $password,
+            'vhost' => $vhost,
+            'port' => $this->port
+        ];
+    }
+
+    /**
+     * Method responsible to open the connection with the broker.
+     *
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     *
+     * @throws \Exception
+     */
+    public function connect($shuffle = true)
+    {
+        if ($shuffle) {
+            $this->shuffleConnections();
+        }
+
+        try {
+            $this->addConnection(AMQPStreamConnection::create_connection($this->connectionsData, [
+                'insist' => true,
+                'login_method' => 'AMQPLAIN',
+                'locale' => 'en_UK',
+                'connection_timeout' => 60.0,
+                'read_write_timeout' => 50.0,
+                'context' => null,
+                'keepalive' => true,
+                'heartbeat' => 0,
+            ]));
```

Comment:
This information should be added when setting up the driver, so that without touching the code of this bundle we can change the options

Reply:
I removed the extra options to configure the connection. If in the future we see the need to implement something related to this, we can do some implementation here.